src/tracer.js

```diff
@@ -316,8 +316,14 @@ function sendCurrentTrace(traceSender) {
         platform: tracerObj.trace.getPlatform(),
     };
 
-
-    if (JSON.stringify(traceJson).length >= consts.MAX_TRACE_SIZE_BYTES) {
+    let stringifyTraceJson;
+    try {
+        stringifyTraceJson = JSON.stringify(traceJson);
```

Comment:
it looks like json.stringify is used in more locations than this. You could create a util or wrap system JSON. also, is a failure that prevents a trace from being sent worth a WARN level?

Reply:
Thanks for your comment! I thought about it as well, but try/catch is a too expensive operation and we prefer not to do that every time.
About the JSON.stringify calls you mentioned - We just deleting properties form traceJson object (has already passed try/catch).
Also, we are going to update this trimming system soon :)